Feature/reusable workflows

gqcorneby opened the pull request:
### :pushpin: References

-   **Issue:** Closes #
https://app.clickup.com/t/86983mnt4
https://app.clickup.com/t/86981nnba

### :memo: Implementation
#### original workflows are from:
- app-test (main) and codeql from metadata synchronization
- bundlemon from user extended app

#### updates
- created a master flow to call all 3 existing flows. Simply add new workflows here and all caller repos will have the update
- updated each workflow to use .nvmrc  for the node version.
- I tried adding inputs (language and build_mode for codeql) that callers can populate and we can add more if ever we want more flexibility
- For bundlemon, I also placed a default  .bundlemonrc  in the workflow repo so we don't have to put this config file for each repo if it'll contain the same thing. If ever, the workflow checks the caller repo first before using the default one in the workflow repo. Overwriting the config inside the repo should be supported but I haven't tested it yet.

- Potentially, if there are projects that require a more customized flow, maybe we can also store them in this centralized repo instead and have the project call it's corresponding workflow

### :video_camera: Screenshots/Screen capture

### :fire: Testing

Commit:
make reusable workflows. add master

Commit:
Merge branch 'master' into feature/reusable-workflows

gqcorneby commented:
I've encountered this error in uhcw which is related to this PR https://github.com/EyeSeeTea/metadata-synchronization/pull/987

![image](https://github.com/user-attachments/assets/938775b1-9a97-452c-9557-aaa77c3acd69)

Given this, I was wondering if we could also use this repo to centralize workflows beyond just generic apps. If UHCW doesn’t require the master workflow, we could consider:
1. Having it call the app-test workflow from this repo.
2. Creating `<repo>.yml` files specifically for UHCW (assuming we rename shared workflows to `shared-<name>.yml` for clarity).

Let me know your thoughts!

tokland commented on .github/workflows/app-test.yml:
```diff
@@ -1,35 +1,35 @@
         uses: actions/setup-node@v3
         with:
-          node-version: "16.14.0"
+          node-version-file: ".nvmrc"
```
Much nicer this way

tokland reviewed (approved):
Untested, but the configurations look ok to me.

Commit:
require app test before codeql and bundlemon to avoid running unnecessary jobs if app-test fails

Commit:
revert bundlemon filename

Commit:
add branch with default "master" for easier testing

Commit:
creat test-master workflow

Commit:
remove dynamic branch from master

Commit:
test master creation

Commit:
fix sed

Commit:
empty test-master

Commit:
use "test-master" in test-master-create

Commit:
fix test-master-create

Commit:
fix test-master-create

Commit:
remove test-master-create. use script instead

Commit:
bundlemon workflow name

Commit:
update app-test versions

Commit:
testing flow

Commit:
use branch bundlemonrc

Commit:
use action ref instead of github ref

Commit:
reorder bundlemon to first look for .bundlemonrc before yarn install build

Commit:
use relative files instead of hardcoding @master in master workflow

Commit:
update test

Commit:
get branch bundlemonrc instead of fixed to master

Commit:
use action path

Commit:
pass branch name

Commit:
add bundlemonrc_branch_name in test-master

Commit:
remove test-master and generator. relative paths are enough except for downloading bundlemonrc

gqcorneby commented:
Noting updates after review:
1. in master.yml, update the workflow calls to use relative path instead of absolute git path with hard coded `@master` branch. This makes it easier to test feature branches
2. To test updates in bundlemonrc in specific branches, I couldn't figure out how to get the workflow's branch from the caller repo so I added `bundlemonrc_branch_name` input from master to specify where to get the file from
3. updated bundlemon workflow to check bundlemonrc first before all the other actions. fail when missing
